cost/google/schedule_instance/google_schedule_instance.pt

```diff
@@ -0,0 +1,638 @@
+    
+    $next_start=split($next_start, "T")
+    $next_start[1]=split($next_start[1],"-")
+    $next_start[1]=join($next_start[1], "m")
+    $next_start[1]=split($next_start[1],":")
+    $next_start[1]=join($next_start[1], "-")
+    $next_start[1]=split($next_start[1], "+")
+    $next_start[1]=join($next_start[1], "p")
+    $next_start=join($next_start, "t")
+	
+    $next_stop=split($next_stop,"T")
+    $next_stop[1]=split($next_stop[1],"-")
+    $next_stop[1]=join($next_stop[1], "m")
+    $next_stop[1]=split($next_stop[1],":")
+    $next_stop[1]=join($next_stop[1], "-")
+    $next_stop[1]=split($next_stop[1], "+")
+    $next_stop[1]=join($next_stop[1], "p")
+    $next_stop=join($next_stop, "t")
+	
+    call get_labelfingerprint($item) retrieve $labelFingerprint, $labels_obj
+	
+    $labels_obj['schedule']=$param
+    $labels_obj['next_start']=$next_start
+    $labels_obj['next_stop']=$next_stop
+    call sys_log('> ' + $item['id'] + ': Updating schedule Tag ...', to_s($item))
+    sub on_error: handle_error($response) do
+      $response= http_post(
+        auth: $$google_auth,
+        url: join([$item['selfLink'], "/setLabels"]),
+        headers:{
+          "cache-control": "no-cache",
+          "content-type": "application/json"
+        },
+        body:{
+          "labels" : $labels_obj,
+          "labelFingerprint" : $labelFingerprint
+        }
+      )
+      $all_response << $response
+    end
+  end
+end
+
+define window_active($start_hour, $start_minute, $start_rule, $stop_hour, $stop_minute, $stop_rule, $tz) return $window_active, $next_start, $next_stop do
+  $params = {
+    verb: 'post',
+    host: 'bjlaftw4kh.execute-api.us-east-1.amazonaws.com',
+    https: true,
+    href: '/production',
+    headers:{
+      'content-type': 'application/json'
+    },
+    body: {
+      'start_hour': $start_hour,
+      'start_minute': $start_minute,
+      'start_rule': $start_rule,
+      'stop_minute': $stop_minute,
+      'stop_hour': $stop_hour,
+      'stop_rule': $stop_rule,
+      'tz': $tz
+    }
+  }
+  $response = http_request($params)
+  $body = $response['body']
+  call sys_log('window active $body', to_s($body))
+
+  $window_active = to_b($body['event_active'])
+  $next_start   = $body['next_start']
+  $next_stop    = $body['next_stop']
+end
+
+define get_labelfingerprint($instance) return $labelFingerprint, $labels_obj do
+  $response = http_get(
+    auth: $$google_auth,
+    url: $instance['selfLink'],
+    headers:{
+      "cache-control": "no-cache",
+      "content-type": "application/json"
+    }
+  )
+  $body = $response['body']
+  $labelFingerprint   = $body['labelFingerprint']
+  $labels_obj= $body['labels']
+end
+
+define get_tzlist() return $timezones do
+  $params = {
+    verb: 'get',
+    host: 'raw.githubusercontent.com',
+    https: true,
+    href: '/rightscale/policy_templates/POL-352-Google-Schedule-Instance/data/tz_database/timezones_list.json',
```

Comment:
This should probably be in the master branch

Reply:
modified